include/zephyr/arch/riscv/structs.h

```diff
@@ -22,6 +22,15 @@ struct _cpu_arch {
 	atomic_ptr_val_t fpu_owner;
 	uint32_t fpu_state;
 #endif
+#if defined(CONFIG_CPP) && !defined(CONFIG_USERSPACE) &&                                           \
+	!(defined(CONFIG_SMP) || (CONFIG_MP_MAX_NUM_CPUS > 1)) && !defined(CONFIG_FPU_SHARING)
+	/* Empty struct has size 0 in C, size 1 in C++. Force them to be the same. */
+	uint8_t unused_cpp_size_compatibility;
+#endif
 };
 
+#if defined(CONFIG_CPP)
+BUILD_ASSERT(sizeof(struct _cpu_arch) >= 1);
+#endif
```

Comment:
What's the purpose of this? To produce a different compilation error?

Reply:
It makes the build failure compiler agnostic. `gcc` doesn't generate this warning, but `clang` does.